File: string/check-if-string-is-a-prefix-of-array/check-if-string-is-a-prefix-of-array.go
package checkifstringisaprefixofarray

func isPrefixString(s string, words []string) bool {
	check := ""
	for _, c := range words {
		check += c
		if check == s {
			return true
		}
		if len(check) > len(s) {
			return false
		}
	}

	return false
}

File: string/check-if-string-is-a-prefix-of-array/check-if-string-is-a-prefix-of-array.ts
function isPrefixString(s: string, words: string[]): boolean {
    let check = "";

    for (let word of words) {
        check += word;
        if (check === s) {
            return true;
        }
        if(check.length > s.length) {
            return false;
        }
    }

    return false;
};
